app.rb

```diff
@@ -0,0 +1,105 @@
+require 'bundler'
+require 'yaml'
+require 'sinatra/base'
+require "sinatra/multi_route"
+require 'securerandom'
+require 'active_support/core_ext/hash'
+require 'tendrl'
+
+Bundler.require :default, ENV['RACK_ENV'].to_sym
+
+class App < Sinatra::Base
+  register Sinatra::RespondWith
+  register Sinatra::CrossOrigin
+  register Sinatra::MultiRoute
+
+  set :root, File.dirname(__FILE__)
+
+  set :env, ENV['RACK_ENV'] || 'development'
+
+  enable :cross_origin
+
+  configure :development, :test do
+    set :etcd_config, Proc.new {
+      YAML.load_file('config/etcd.yml')[settings.env.to_sym] 
+    }
+  end
+
+  configure :production do
+    set :etcd_config, Proc.new {
+      YAML.load(ERB.new(File.read('config/etcd_prod.yml')).result)[settings.env.to_sym] 
+    }
+  end
+
+  before do
+    Tendrl.sds_config
+  end
+
+  set :etcd, Proc.new {
+    Etcd.client(
+      host: etcd_config[:host],
+      port: etcd_config[:port],
+      user_name: etcd_config[:user_name],
+      password: etcd_config[:password]
+    )
+  }
+
+  before do
```

Comment:
@mkudlej the block is used to set the etcd attributes to the client for connecting. This will read the etcd attributes from etcd.yml file and connect to the local db.